vmaas/webapp/repos.py

```diff
@@ -164,8 +163,12 @@ def process_list(self, api_version, data):  # pylint: disable=unused-argument
                         "cpes": [self.cache.cpe_id2label[cpe_id] for cpe_id in cpe_ids],
                         "third_party": repo_detail[REPO_THIRD_PARTY]
                     }
-                    self._set_updated_packages(repo, repo_id, repoid2errataids, modified_since_dt, show_packages)
-                    repolist.setdefault(label, []).append(repo)
+                    updated_packages = self._get_updated_packages(repo_id, repoid2errataids)
+                    if show_packages:
+                        repo["updated_package_names"] = updated_packages
+                    # Skip repository if there are no changed packages found
+                    if updated_packages or not modified_since_dt:
+                        repolist.setdefault(label, []).append(repo)
```

Comment:
document in apispec that modified_since is not filtering on repo timestamp but it is also excluding repos with no new errata